### PR TITLE
[posix-app] fix diag issues of radio only mode

### PR DESCRIPTION
**Pull request by bukepo**
* don't compute CRC in promiscuous mode
* call `otPlatDiagRadioTransmitDone()` when transmit is done
* don't wait for ACK in promiscuous mode

### CODE REVIEW

File: include/openthread-config-android.h [hunk 1]
```diff
@@ -143,13 +143,13 @@
...
 /* Define to 1 to enable the NCP SPI interface. */
-#define OPENTHREAD_ENABLE_NCP_SPI 1
+#define OPENTHREAD_ENABLE_NCP_SPI 0
 
 /* Define to 1 if using NCP Spinel Encrypter */
 #define OPENTHREAD_ENABLE_NCP_SPINEL_ENCRYPTER 0
 
 /* Define to 1 to enable the NCP UART interface. */
-#define OPENTHREAD_ENABLE_NCP_UART 0
+#define OPENTHREAD_ENABLE_NCP_UART 1
```
Comment (jwhui) [inline include/openthread-config-android.h]: Is this change intentional?

Comment (bukepo) [reply include/openthread-config-android.h]: Yes, it is. This configures how the host side provide NCP tunnel for wpantund. I changed this by mistake in last PR.
